packages/compass-saved-aggregations-queries/src/stores/aggregations-queries-items.ts

```diff
@@ -68,7 +68,7 @@ const reducer: Reducer<State, Actions | EditItemActions | DeleteItemActions> = (
         return state;
       }
       const updatedItem =
-        item.type === 'query'
+        item.type === 'query' || item.type === 'updatemany'
```

Comment:
Drive-by. Seemed like dangerous code. Maybe better to check for item.query rather?

Reply:
Tbh the part that looks particularly dangerous here is the assertions below, this should've been caught by typescript if not for them, I wonder why we assert there

Reply:
Can't tell how it would know:

```
type EditItemUpdatedAction = {
  type: ActionTypes.EditItemUpdated;
  id: string;
  payload: FavoriteQuery | SavedPipeline;
};
```

Reply:
I suppose it could be two things ORed together? Let me try.

Reply:
Actually no. That is gonna be non-trivial.

Reply:
Ooh this seems to work:
```javascript
      const updatedItem =  'name' in action.payload
          ? mapAggregationToItem(action.payload)
          : mapQueryToItem(action.payload);
```